src/nnet3/discriminative-supervision.cc

```diff
@@ -400,49 +400,36 @@ void DiscriminativeSupervisionSplitter::ComputeLatticeScores(const Lattice &lat,
 }
 
 void AppendSupervision(const std::vector<const DiscriminativeSupervision*> &input,
-    bool compactify,
-    std::vector<DiscriminativeSupervision> *output_supervision) {
+    DiscriminativeSupervision *output_supervision) {
   KALDI_ASSERT(!input.empty());
   int32 num_inputs = input.size();
   if (num_inputs == 1) {
-    output_supervision->resize(1);
-    (*output_supervision)[0] = *(input[0]);
+    (*output_supervision) = *(input[0]);
     return;
   }
-  std::vector<bool> output_was_merged;
-  output_supervision->clear();
-  output_supervision->reserve(input.size());
-  for (int32 i = 0; i < input.size(); i++) {
+  (*output_supervision) = *(input[num_inputs-1]);
+  for (int32 i = num_inputs-2; i > -1; i--) {
```

Comment:
here and elsewhere, please put space around the operator -, i.e. 'num_inputs - 2'.  This is how we normally do it.  And I prefer 'i >= 0' to 'i > -1'.